### PR TITLE
Add MinimumStorageDeposit and remove some TODOs

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: sdk/src/types/block/output/rent.rs [hunk 1]
```diff
@@ -140,3 +148,44 @@ impl<T: Rent, const N: usize> Rent for [T; N] {
...
+
+    pub fn with_storage_deposit_return(mut self, return_address: Address) -> Result<Self, Error> {
+        self.builder = self
+            .builder
+            .add_unlock_condition(StorageDepositReturnUnlockCondition::new(
+                return_address,
+                Output::AMOUNT_MIN,
+                self.token_supply,
+            )?)
+            .add_unlock_condition(ExpirationUnlockCondition::new(return_address, 1)?);
```
Comment (thibault-martinez) [inline sdk/src/types/block/output/rent.rs]: Should we add it automatically or have another with_ method ?

Comment (Thoralf-M) [reply sdk/src/types/block/output/rent.rs]: Should be separated, it's not always wanted

File: sdk/src/types/block/output/rent.rs [hunk 2]
```diff
@@ -140,3 +148,44 @@ impl<T: Rent, const N: usize> Rent for [T; N] {
         self.iter().map(|elem| elem.weighted_bytes(config)).sum()
     }
 }
+
+pub struct MinimumStorageDeposit {
+    config: RentStructure,
+    token_supply: u64,
+    builder: BasicOutputBuilder,
```
Comment (Thoralf-M) [inline sdk/src/types/block/output/rent.rs]: Maybe call the whole struct `MinimumStorageDepositBasicOutput`? For other output types it's different
Or it should be possible to provide the type of output? (probably not worth it right now if we don't use it anywhere)

Comment (thibault-martinez) [reply sdk/src/types/block/output/rent.rs]: Are we ever interested in non basic though? We only use this for returns

Comment (Thoralf-M) [reply sdk/src/types/block/output/rent.rs]: From the name it sounds like a generic thing and not only for this specific case
If we don't want it generic, then I would make it pub(crate) and maybe add a comment above that it's used for storage deposit returns

File: sdk/src/types/block/output/rent.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/types/block/output/rent.rs]: Done

File: sdk/src/types/block/output/rent.rs [hunk 3]
```diff
@@ -140,3 +148,50 @@ impl<T: Rent, const N: usize> Rent for [T; N] {
...
+            .builder
+            .add_unlock_condition(StorageDepositReturnUnlockCondition::new(
+                return_address,
+                Output::AMOUNT_MIN,
+                self.token_supply,
+            )?);
+        Ok(self)
+    }
+
+    pub fn with_expiration(mut self, return_address: Address) -> Result<Self, Error> {
```
Comment (Thoralf-M) [inline sdk/src/types/block/output/rent.rs]: Why accept an address here and not also do `Address::from(Ed25519Address::from([0; Ed25519Address::LENGTH]))`?

Comment (thibault-martinez) [reply sdk/src/types/block/output/rent.rs]: Yeah you are right, blind copy paste, will change

Comment (thibault-martinez) [reply sdk/src/types/block/output/rent.rs]: Done

File: sdk/src/types/block/output/rent.rs (hunk 2 above)
Comment (thibault-martinez) [reply sdk/src/types/block/output/rent.rs]: I'll just rename it

Comment (thibault-martinez) [reply sdk/src/types/block/output/rent.rs]: Done

File: sdk/src/types/block/output/rent.rs [hunk 4]
```diff
@@ -140,3 +148,51 @@ impl<T: Rent, const N: usize> Rent for [T; N] {
...
+    token_supply: u64,
+    builder: BasicOutputBuilder,
+}
+
+impl MinimumStorageDepositBasicOutput {
+    pub fn new(config: RentStructure, token_supply: u64) -> Self {
+        Self {
+            config,
+            token_supply,
+            builder: BasicOutputBuilder::new_with_amount(Output::AMOUNT_MIN).add_unlock_condition(
```
Comment (DaughterOfMars) [inline sdk/src/types/block/output/rent.rs:163]: Isn't there a `new_with_min_storage_deposit` variant?

Comment (thibault-martinez) [reply sdk/src/types/block/output/rent.rs:163]: There is, but in this case the result is the same and `new_with_min_storage_deposit` induces a clone of the output

Comment (DaughterOfMars) [reply sdk/src/types/block/output/rent.rs:163]: boo

File: bindings/python/examples/wallet/destroy_alias.py
Comment (Thoralf-M) [inline bindings/python/examples/wallet/destroy_alias.py:1]: Actually can you delete this file?
We have how_tos/alias/destroy.py already

Comment (thibault-martinez) [reply bindings/python/examples/wallet/destroy_alias.py:1]: Yes sir

Comment (thibault-martinez) [reply bindings/python/examples/wallet/destroy_alias.py:1]: Done

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**